src/util/newconfig/ConfigDefinition.hpp

```diff
@@ -229,4 +279,108 @@ class ClioConfigDefinition {
+     {"database.cassandra.username", ConfigValue{ConfigType::String}.optional()},
+     {"database.cassandra.password", ConfigValue{ConfigType::String}.optional()},
+     {"database.cassandra.certfile", ConfigValue{ConfigType::String}.optional()},
+     {"allow_no_etl", ConfigValue{ConfigType::Boolean}.defaultValue(false)},
+     {"etl_sources.[].ip", Array{ConfigValue{ConfigType::String}.optional().withConstraint(validateIP)}},
+     {"etl_sources.[].ws_port", Array{ConfigValue{ConfigType::String}.optional().withConstraint(validatePort)}},
+     {"etl_sources.[].grpc_port", Array{ConfigValue{ConfigType::String}.optional().withConstraint(validatePort)}},
+     {"forwarding.cache_timeout",
+      ConfigValue{ConfigType::Double}.defaultValue(0.0).withConstraint(validatePositiveDouble)},
+     {"forwarding.request_timeout",
+      ConfigValue{ConfigType::Double}.defaultValue(10.0).withConstraint(validatePositiveDouble)},
+     {"rpc.cache_timeout", ConfigValue{ConfigType::Double}.defaultValue(0.0).withConstraint(validatePositiveDouble)},
+     {"num_markers", ConfigValue{ConfigType::Integer}.optional().withConstraint(validateNumMarkers)},
+     {"dos_guard.whitelist.[]", Array{ConfigValue{ConfigType::String}}},
+     {"dos_guard.max_fetches", ConfigValue{ConfigType::Integer}.defaultValue(1000'000u).withConstraint(validateUint32)},
+     {"dos_guard.max_connections", ConfigValue{ConfigType::Integer}.defaultValue(20u).withConstraint(validateUint32)},
+     {"dos_guard.max_requests", ConfigValue{ConfigType::Integer}.defaultValue(20u).withConstraint(validateUint32)},
+     {"dos_guard.sweep_interval",
+      ConfigValue{ConfigType::Double}.defaultValue(1.0).withConstraint(validatePositiveDouble)},
+     {"workers",
+      ConfigValue{ConfigType::Integer}.defaultValue(std::thread::hardware_concurrency()).withConstraint(validateUint32)
+     },
+     {"server.ip", ConfigValue{ConfigType::String}.withConstraint(validateIP)},
+     {"server.port", ConfigValue{ConfigType::Integer}.withConstraint(validatePort)},
+     {"server.max_queue_size", ConfigValue{ConfigType::Integer}.defaultValue(0).withConstraint(validateUint32)},
+     {"server.local_admin", ConfigValue{ConfigType::Boolean}.optional()},
+     {"server.admin_password", ConfigValue{ConfigType::String}.optional()},
+     {"server.processing_policy",
+      ConfigValue{ConfigType::String}.defaultValue("parallel").withConstraint(validateProcessingPolicy)},
+     {"server.parallel_requests_limit", ConfigValue{ConfigType::Integer}.optional()},
+     {"server.ws_max_sending_queue_size", ConfigValue{ConfigType::Integer}.defaultValue(1500)},
+     {"server.__ng_web_server", ConfigValue{ConfigType::Boolean}.defaultValue(false)},
+     {"prometheus.enabled", ConfigValue{ConfigType::Boolean}.defaultValue(true)},
+     {"prometheus.compress_reply", ConfigValue{ConfigType::Boolean}.defaultValue(true)},
+     {"io_threads", ConfigValue{ConfigType::Integer}.defaultValue(2).withConstraint(validateUint16)},
```

Comment:
If we set io_threads 1, does Clio exit with error as it does previously?

Reply:
@cindyyan317  I set `io_threads` to 1 in the old config and it seems to work perfectly fine. We can maybe have a look?

Reply:
Forgive me. https://github.com/XRPLF/clio/blob/a7074dbf0f2b63dd76fdd48b1879b468513a78d0/src/app/ClioApplication.cpp#L107. I think i should ask if we set it 0, does Clio exit with error.

Reply:
Thank you for the catch 🙏. The constraint for `io_threads` needs to be changed